rotkehlchen/assets/spam_assets.py

```diff
@@ -79,12 +79,18 @@ def update_spam_assets(db: 'DBHandler', assets_info: list[dict[str, Any]]) -> in
     # and in that case query_token_spam_list add it to the global DB
     with db.conn.read_ctx() as cursor:
         ignored_asset_ids = db.get_ignored_asset_ids(cursor)
-    assets_added = 0
+
+    write_data = []
     for token in spam_tokens:
         if token.identifier in ignored_asset_ids:
             continue
 
-        with db.user_write() as write_cursor:
-            db.add_to_ignored_assets(write_cursor=write_cursor, asset=token)
-        assets_added += 1
+        write_data.append(('ignored_asset', token.identifier))
+
+    with db.user_write() as write_cursor:
```

Comment:
This is a an important adjustment of this PR. Using db.add_to_ignored_assets in a loop should be avoided as it:

1. Makes as many writes as there is assets
2. Takes an asset and calls many asset specific functions while you just need the identifier